<h1 style='background: black; border:1; color: white'><center>1. 我国电力及其碳排放概况</center></h1>

<h2 style='background: black; border:1; color: white'><center>1.1 2005—2016年我国电力生产状况</center></h2>

In [ ]:
import pandas as pd
import numpy as np

<h3 style='background: black; border:1; color: white'><center>1.1.1 数据展示</center></h3>

In [100]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings("ignore")

In [12]:
hv_lines = pd.read_excel(r'2_1.xlsx')
hv_lines

年份    电力生产量        水电        火电      核电      风电    其他电力
0   2005  25002.6   3970.20  20473.40   530.9     0.0   28.10
1   2006  28657.3   4357.90  23696.00   548.4     0.0   55.00
2   2007  32815.5   4852.64  27229.33   621.3     0.0  112.23
3   2008  34668.8   5851.90  27900.80   683.9     0.0  232.20
4   2009  37146.5   6156.40  29827.80   701.3     0.0  461.00
5   2010  42071.6   7221.70  33319.30   738.8   446.2  345.60
6   2011  47130.2   6989.50  38337.00   863.5   703.3  236.90
7   2012  49875.5   8721.10  38928.10   973.9   959.8  292.60
8   2013  54316.4   9202.90  42470.10  1116.1  1412.0  115.30
9   2014  56495.8  10643.40  42686.50  1325.4  1560.8  279.70
10  2015  58146.0  11303.00  42842.00  1708.0  1858.0  435.00
11  2016  61322.0  11840.00  44371.00  2133.0  2371.0  607.00

<img src="pic/2_1.png" height=50%>

<h3 style='background: black; border:1; color: white'><center>1.1.2 结论</center></h3>

1. 与 2005 年相比，2016 年我国**发电总量**增加36422.30 亿 kW·h，增长率 145.67% ，年均增长率 8.51%。
2. 我国电力生产以**火电**为主，其占比由 81.89% 降为72.24%，年均增长率 7.28% ，2013 年后其增长明显趋缓。
3. **水电、核电等非化石电力生产量**增长276.54% ，比重提高了9.65个百分点，但仍明显低于火电。

<h2 style='background: black; border:1; color: white'><center>1.2 我国电力碳排放量及碳排放强度</center></h2>

<h3 style='background: black; border:1; color: white'><center>1.2.1 数据展示</center></h3>

In [13]:
carbon= pd.read_excel(r'2_2.xlsx')
carbon

年份       电力碳排放量     电力碳排放强度
0   2005  2116.374808  846.460000
1   2006  2222.193548  837.995400
2   2007  2463.169221  796.095630
3   2008  2571.440395  788.134674
4   2009  2706.779363  796.016020
5   2010  3007.532626  788.055860
6   2011  3538.273678  780.175302
7   2012  3724.498608  764.571796
8   2013  4133.280000  741.634642
9   2014  3941.495808  674.887524
10  2015  3768.858292  654.640898
11  2016  3979.207914  647.800000

<img src="pic/2_2.png" >

<h3 style='background: black; border:1; color: white'><center>1.2.2 结论</center></h3>

**电力碳排放量**在2013年达到一个峰值，为4133.28Mt，相当于2005年排放量的195.30%;2014年、2015年电力碳排放量均比上年降低，2016小幅增长，但仍低于 2013年的排放量。2016年碳排放量比2005年增长了88.02% ，年均增长5.91%。而**电力碳强度**则逐年降低，从 846.46g /(kW·h)下降至 647.80g /(kW·h)，年均降低2.40%。

<h1 style='background: black; border:1; color: white'><center>2. 电力碳排放影响因素的贡献</center></h1>

<h2 style='background: black; border:1; color: white'><center>2.1 数据展示</center></h2>

在发电环节，按照电源体系的特点将强度因素具体化为**1.碳排放系数、2.火电能耗强度**两项因素，将结构因素具体化为**3.火力发电能源结构、4.电力结构**两项因素。

在输电环节，选择强度因素并具体化为**5.发用电比因素**。

在用电环节，考虑电力消费的普遍性选择了**6.经济规模和7.用电强度因素**。

In [122]:
e_carbon= pd.read_excel(r'3_1.xlsx')
e_carbon

年份     1.电力碳排放量   1.电力碳排放强度   2.煤炭   2.石油  2.天然气   2.热能   3.能源消费总量  \
0   2005  2116.374808  846.460000  0.724  0.178  0.024  0.074    2613.69   
1   2006  2222.193548  837.995400  0.724  0.175  0.027  0.074    2864.67   
2   2007  2463.169221  796.095630  0.725  0.170  0.030  0.075  311442.00   
3   2008  2571.440395  788.134674  0.715  0.167  0.034  0.084    3206.11   
4   2009  2706.779363  796.016020  0.716  0.164  0.035  0.085    3361.26   
5   2010  3007.532626  788.055860  0.692  0.174  0.040  0.094    3606.48   
6   2011  3538.273678  780.175302  0.702  0.168  0.046  0.084    3870.43   
7   2012  3724.498608  764.571796  0.685  0.170  0.048  0.097    4021.38   
8   2013  4133.280000  741.634642  0.674  0.171  0.053  0.102    4169.13   
9   2014  3941.495808  674.887524  0.656  0.174  0.057  0.113    4258.06   
10  2015  3768.858292  654.640898  0.637  0.183  0.059  0.121    4299.05   
11  2016  3979.207914  647.800000  0.620  0.183  0.064  0.133    4360.00   

      3.火力发电量   3.火电能耗强度  ...  4.电力生产总量    4.电力强度  5.电力生产总量  5.电力终端消费量  \
0     204.734  12.766272  ...   25002.6  0.008189   25002.6    23234.0   
1     236.960  12.089256  ...   28657.3  0.008269   28657.3    26729.1   
2   27229.300  11.437753  ...   32815.5  0.829769   32815.5    30650.1   
3     279.008  11.491104  ...   34668.8  0.008048   34668.8    32403.5   
4     298.278  11.268883  ...   37146.5  0.008030   37146.5    34773.9   
5     333.193  10.823997  ...   42071.6  0.007920   42071.6    39366.0   
6     383.370  10.095808  ...   47130.2  0.008134   47130.2    44300.2   
7     389.281  10.330276  ...   49875.5  0.007805   49875.5    46866.5   
8     424.701   9.816624  ...   54316.4  0.007819   54316.4    51062.7   
9     426.865   9.975191  ...   56495.8  0.007556   56495.8    53283.8   
10    428.419  10.034686  ...   58146.0  0.007368   58146.0    55032.0   
11    443.710   9.826238  ...   61332.0  0.007235   61332.0    58142.0   

      5.发用电比     6.GDP   6.能源消费总量  7.电力终端消费量     7.GDP    7.用电强度  
0   1.076121  187318.9    2613.69    23234.0  187318.9  0.124034  
1   1.072139  219438.5    2864.67    26729.1  219438.5  0.121807  
2   1.070649  270092.3  311442.00    30650.1  270092.3  0.113480  
3   1.069909  319244.6    3206.11    32403.5  319244.6  0.101501  
4   1.068229  348517.7    3361.26    34773.9  348517.7  0.099777  
5   1.068729  412119.3    3606.48    39366.0  412119.3  0.095521  
6   1.063882  487940.2    3870.43    44300.2  487940.2  0.090790  
7   1.064204  538580.0    4021.38    46866.5  538580.0  0.087019  
8   1.063720  592963.2    4169.13    51062.7  592963.2  0.086114  
9   1.060281  643563.1    4258.06    53283.8  643563.1  0.082795  
10  1.056585  688858.2    4299.05    55032.0  688858.2  0.079889  
11  1.054866  746395.1    4360.00    58142.0  746395.1  0.077897  

[12 rows x 21 columns]

<h2 style='background: black; border:1; color: white'><center>2.2 模型计算</center></h2>

In [129]:
#LMDI法 
def LMDI(data,tt,t0):
    def Delta_XX(yt,y0,xt,x0):
        def L(yt,y0):
            if yt == y0:
                return 0
            else:
                return (yt-y0)/(np.log(yt) - np.log(y0))
        return L(yt,y0)*np.log(xt/x0)
    Delta_y = data[tt,0]-data[t0,0]
    Delta_x = []
    for i in range(1,np.shape(data)[1]):
        Delta_x.append( Delta_XX(data[tt,0],data[t0,0],data[tt,i], data[t0,i]) )
    return Delta_y,Delta_x

#分解
def dt(b,d):
    for i in range(0,np.shape(b)[0]-1):
        c_0=LMDI(b,i+1,i)
        c_1=np.array(c_0)
        c_2=np.array(c_1[1])
        d[i]=c_2[0]
    return d

a_1=pd.read_excel(r'3_1_1.xlsx')
a_2=pd.read_excel(r'3_1_2.xlsx')
a_3=pd.read_excel(r'3_1_3.xlsx')
a_4=pd.read_excel(r'3_1_4.xlsx')
a_5=pd.read_excel(r'3_1_5.xlsx')
a_6=pd.read_excel(r'3_1_6.xlsx')
a_7=pd.read_excel(r'3_1_7.xlsx')

a=a_1.iloc[:,1:3]
b=a.values
d=[0 for i in range(np.shape(b)[0]-1)]
b_1=dt(b,d)

a=a_2.iloc[:,1:5]
b=a.values
d=[0 for i in range(np.shape(b)[0]-1)]
b_2=dt(b,d)

a=a_3.iloc[:,1:4]
b=a.values
d=[0 for i in range(np.shape(b)[0]-1)]
b_3=dt(b,d)

a=a_4.iloc[:,1:4]
b=a.values
d=[0 for i in range(np.shape(b)[0]-1)]
b_4=dt(b,d)

a=a_5.iloc[:,1:4]
b=a.values
d=[0 for i in range(np.shape(b)[0]-1)]
b_5=dt(b,d)

a=a_6.iloc[:,1:3]
b=a.values
d=[0 for i in range(np.shape(b)[0]-1)]
b_6=dt(b,d)

a=a_7.iloc[:,1:4]
b=a.values
d=[0 for i in range(np.shape(b)[0]-1)]
b_7=dt(b,d)

<h2 style='background: black; border:1; color: white'><center>2.3 结果</center></h2>

In [132]:
pd.read_excel(r'3_2.xlsx')

年份         总效应   排放系数  火力发电能源结构  火电能耗强度        电力结构       发用电比  \
0   2005—2006   49.784931   0.00     -4.23  -32.10    5.744569  -2.184244   
1   2006—2007  243.243367   0.00     -3.81 -163.73   10.328410  -4.109126   
2   2007—2008  162.092059   0.00      0.17   25.97 -170.638848  -3.858027   
3   2008—2009  229.624376   0.00     -5.21   53.13   -6.105083  -4.274944   
4   2009—2010  693.262131  29.17     79.45  155.98  -53.468201   1.812221   
5   2010—2011   10.486548   0.00      2.87   -7.51    2.883175  -0.490166   
6   2011—2012  188.746238   0.00      3.12  111.39 -200.450970   1.465102   
7   2012—2013   69.851377  -5.77     16.88  -83.03    2.910247  -0.740486   
8   2013—2014 -129.685382  -0.21     15.44 -212.23 -453.729961 -42.885599   
9   2014—2015  -41.408869  -0.58     -3.98  -94.75 -400.841905 -55.635596   
10  2015—2016   28.164177   0.15      4.87  -34.51  -30.044271  -2.677775   

          用电强度         经济规模  
0   -10.676526    93.231132  
1  -209.278240   613.842323  
2  -622.584745   933.033679  
3   -46.607983   238.692387  
4  -168.793159   649.111270  
5    -5.477149    18.210688  
6  -205.856799   479.078905  
7   -17.004531   156.606147  
8  -520.641842  1084.572020  
9  -569.355958  1083.734590  
10  -41.504654   131.880877

<img src="pic/3_1.png" >

<h2 style='background: black; border:1; color: white'><center>2.4 结论</center></h2>

1. 第一个典型年份为2006年，该年电力碳排放的增量明显低于其后两年。
2. 第二个典型年份为2010年，该年电力碳排放的增量远高于其他年份。
3. 第三个典型年份为2014年，该年电力碳排放首次减少，形成了一个拐点。

<h1 style='background: black; border:1; color: white'><center>3. 主要影响因素的各省差异</center></h1> 

<h2 style='background: black; border:1; color: white'><center>3.1 我国30个省份 2016 年人均 GDP及近3年GDP 均增速</center></h2>

<h3 style='background: black; border:1; color: white'><center>3.1.1 数据展示</center></h3>

In [14]:
pgdp= pd.read_excel(r'3_3.xlsx')
pgdp

省份   人均GDP  GDP增速%
0    天津  115613    5.12
1    上海  114690    7.10
2    北京  113731    6.80
3    江苏   95394    7.20
4    浙江   83923    7.30
5    福建   74288    8.10
6   内蒙古   74204    5.20
7    广东   73290    7.20
8    山东   68049    7.10
9    重庆   58199    8.60
10   湖北   55191    7.90
11   吉林   54073    5.30
12   陕西   50528    7.80
13   辽宁   50292    7.80
14   宁夏   47157    7.30
15   湖南   46063    7.90
16   海南   44396    7.80
17   青海   43750    7.40
18   河北   42866    7.80
19   河南   42363    7.80
20   新疆   40466    7.20
21  黑龙江   40362    8.00
22   江西   40220    8.80
23   四川   39835    8.00
24   安徽   39254    8.30
25   广西   38042    7.00
26   山西   35285    6.00
27   贵州   33242   10.00
28   云南   31358    9.10
29   甘肃   27508    5.90

<img src="pic/3_4.png" >

<h3 style='background: black; border:1; color: white'><center>3.1.2 结论</center></h3>

1. 以这2项指标的 **中位数**为轴，形成了 **4个象限**。
2. 若仅从促进全国发展平衡的角度看，GDP增速和人均GDP都较高的**第一象限**省份应首先防止GDP过快增长
3. **第二象限**、**第三象限**的省份人均GDP偏低，需在满足合理增速的同时，努力促进**经济增长**与**碳排放增长**脱钩。

<h2 style='background: black; border:1; color: white'><center>3.2 我国30个省份2016年火电占比及火电能耗强度</center></h2>

<h3 style='background: black; border:1; color: white'><center>3.2.1 数据展示</center></h3>

In [23]:
fea= pd.read_excel(r'3_4.xlsx')
fea

省份  火电能耗强度     火电占比%
0    云南    3.82  0.123500
1   内蒙古    3.81  0.843400
2    吉林    3.79  0.791200
3    辽宁    3.51  0.791500
4   黑龙江    3.42  0.887800
5    新疆    3.25  0.817800
6    山西    3.22  0.935400
7    河北    3.17  0.898100
8    青海    3.13  0.363800
9    宁波    3.05  0.822100
10   河南    3.05  0.985600
11   福建    3.02  0.432500
12   广西    2.99  0.419600
13   甘肃    2.98  0.598600
14   天津    2.91  0.990514
15   贵州    2.89  0.622500
16   上海    2.89  0.991000
17   湖南    2.85  0.542600
18   江西    2.83  0.791400
19   江苏    2.82  0.972500
20   四川    2.81  0.178600
21   重庆    2.81  0.643400
22   山东    2.81  0.999100
23   安徽    2.78  0.963500
24   海南    2.73  0.701200
25   广东    2.71  0.708900
26   陕西    2.67  0.890200
27   湖北    2.66  0.409200
28   浙江    2.62  0.785600
29   北京    2.18  0.985000

<img src="pic/3_5.png">

<h3 style='background: black; border:1; color: white'><center>3.2.2 结论</center></h3>

1. 亟待调整的是内蒙古、黑龙江及新疆、山西、宁夏、河北等**第一象限**省份，这些省份火电能耗强度，火电占比过高，应当大力提高火力发电效率，尽快使其赶超全国平均水平，并要因地制宜发展非化石电力，尽量降低火电比重。
2. **第二象限**中云南、吉林、辽宁三省的火电能耗强度大幅高于中位数，也要着力促其降低。

<h2 style='background: black; border:1; color: white'><center>3.3 我国30个省份2016年单位GDP用电量及火电占比</center></h2>

<h3 style='background: black; border:1; color: white'><center>3.3.1 数据展示</center></h3>

In [26]:
ydl= pd.read_excel(r'3_5.xlsx')
ydl

省份     火电占比%  单位GDP电耗
0    云南  0.123500     13.5
1   内蒙古  0.843400     16.6
2    吉林  0.791200      5.5
3    辽宁  0.791500      9.9
4   黑龙江  0.887800      6.3
5    新疆  0.817800     29.1
6    山西  0.935400     16.1
7    河北  0.898100     11.7
8    青海  0.363800     37.2
9    宁波  0.822100     45.5
10   河南  0.985600     11.1
11   福建  0.432500      9.5
12   广西  0.419600     10.1
13   甘肃  0.598600     15.3
14   天津  0.990514      6.1
15   贵州  0.622500     16.5
16   上海  0.991000      7.6
17   湖南  0.542600      7.4
18   江西  0.791400      9.7
19   江苏  0.972500      7.5
20   四川  0.178600      8.9
21   重庆  0.643400      6.8
22   山东  0.999100      9.9
23   安徽  0.963500     11.4
24   海南  0.701200      9.9
25   广东  0.708900      9.2
26   陕西  0.890200     11.5
27   湖北  0.409200      9.2
28   浙江  0.785600     10.2
29   北京  0.985000      6.2

<img src="pic/3_6.png">

<h3 style='background: black; border:1; color: white'><center>3.3.2 结论</center></h3>

1. 迫切需要调整的是**第一象限**的省份，尤其是宁夏、新疆、内蒙古、山西，其用电强度大幅高于中位数，火电占比也过高，应把降低单位 GDP 耗电量作为主攻方向，同时尽量降低火电比重。
2. 青海的用电强度达到中位数的3倍以上，贵州、甘肃的用电强度为中位数的1.5倍以上，也应当把降低单位GDP电耗作为攻坚目标。
3. 上海、天津、山东、北京火电占比均在96%以上，且经济实力较强，应尽量挖掘**各种非化石电力**的潜力。